platform/admin/admin-configuration-passwordencryption/src/test/java/org/codice/ddf/admin/configuration/password/encryption/PasswordEncryptingPersistenceManagerTest.java

```diff
@@ -0,0 +1,182 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.configuration.password.encryption;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+
+import org.apache.felix.cm.PersistenceManager;
+import org.codice.ddf.admin.core.api.ConfigurationAdmin;
+import org.codice.ddf.admin.core.api.MetatypeAttribute;
+import org.codice.ddf.admin.core.api.Service;
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.service.metatype.AttributeDefinition;
+
+import ddf.security.encryption.EncryptionService;
+
+public class PasswordEncryptingPersistenceManagerTest {
+
+    private PasswordEncryptingPersistenceManager passwordEncryptingPersistenceManager;
+
+    private EncryptionService encryptionService;
+
+    private PersistenceManager persistenceManager;
+
+    private ConfigurationAdmin configurationAdmin;
+
+    @Before
+    public void setup() throws IOException {
+        persistenceManager = mock(PersistenceManager.class);
+        configurationAdmin = mock(ConfigurationAdmin.class);
+        encryptionService = mock(EncryptionService.class);
+        passwordEncryptingPersistenceManager = new PasswordEncryptingPersistenceManager(
+                persistenceManager,
+                encryptionService,
+                configurationAdmin) {
+            List<Service> getServices(String pid) {
+                return configurationAdmin.listServices("(service.factoryPid=" + pid + ")",
+                        "(service.pid=" + pid + ")");
+            }
+        };
+        Hashtable encrypted = new Hashtable();
+        encrypted.put("service.pid", "pid");
+        encrypted.put("password", "encrypted");
+        encrypted.put("notpassword", "foo");
+        Hashtable noPassword = new Hashtable();
+        noPassword.put("service.factoryPid", "pid.uuid");
+        noPassword.put("random", "stuff");
+        Hashtable nullPassword = new Hashtable();
+        noPassword.put("notstring", new ArrayList<String>());
+        Enumeration enumeration = Collections.enumeration(Arrays.asList(encrypted,
+                noPassword,
+                nullPassword));
+        when(persistenceManager.load("123")).thenReturn(encrypted);
```

Comment:
✏️  Can we make the pids a little more descriptive instead of using `123`, `321`, etc., or use constants?